streams/src/main/java/org/apache/kafka/streams/processor/internals/TaskManager.java

```diff
@@ -1226,7 +1226,9 @@ private void releaseLockedUnassignedTaskDirectories() {
         final Iterator<TaskId> taskIdIterator = lockedTaskDirectories.iterator();
         while (taskIdIterator.hasNext()) {
             final TaskId id = taskIdIterator.next();
-            if (!tasks.contains(id)) {
+            final Set<TaskId> tasksInStateUpdater = stateUpdater != null
```

Comment:
We probably want to do this outside of the loop.

Reply:
Also could we just store a copy of `allTasks()` and use containsKey?

Reply:
> Also could we just store a copy of allTasks() and use containsKey?

`allTasks()` returns the tasks owned by the stream thread. Those do not include tasks owned by the state updater. Indeed, `tasks.contains()` looks up the tasks in the same data structures that `allTasks()` uses. So, using `allTasks()` would not fix the bug.

Reply:
> We probably want to do this outside of the loop.

Good point!

Reply:
I think what you are saying is not correct. Note that `tasks.allTasks()` does not include state-updater owned tasks, but `allTasks()` (in this context, so TaskManger.allTasks) creates a common datastructure of all tasks, that you could use here. Either way, the current implementation is fine with me as well.

Reply:
Ah, a classic misunderstanding! 
Since your proposal simplifies the code, I applied it. Thanks!